third_party/move/move-binary-format/src/access.rs

```diff
@@ -192,6 +192,37 @@ pub trait ModuleAccess: Sync {
             .collect()
     }
 
+    /// Returns an iterator that iterates over all immediate dependencies of the module.
+    ///
+    /// This is more efficient than `immediate_dependencies` as it does not make new
+    /// copies of module ids on the heap.
+    fn immediate_dependencies_iter(
```

Comment:
good catch with interning here! we can unify later.
nit: maybe immediate_dependencies_ref_iter or smt could be more descriptive?

Reply:
Yeah, let's refactor these interface a bit later.